### ISSUE TITLE
Not working on all my Echo Dot 2nd Gen and Echo Plus

### ISSUE THREAD

**Issue by PlayedIn** [thorsten-gehrig/alexa-remote-control#23]
**I can get -lastalexa to work, but fails to speak, or run routines on 4 2nd gen echo dots and 1 echo plus, It doesn't work but there's no error.  Here is the output for an automation (routine) that opens a garage door:**
```
core-ssh:/config/alexa# bash alexa_remote_control.sh -e automation:'Open Door'
setting default device to:
Laundry Room
sending cmd:automation:Open Door to dev:Laundry Room type:A3S5BH2HU6VAYF serial:G090L91182330V20 customerid:A17XZBYWJG1SFN
Running routine: Open Door
HTTP/1.1 0 x-amzn-RequestId: 29ef6941-f371-11e8-8b75-e72497d798ce
Access-Control-Allow-Origin: https://alexa.amazon.com
Content-Encoding: deflate
Vary: Origin
Access-Control-Expose-Headers: x-amzn-dat-gui-client-upgrade,X-Amzn-Error,loginUrl,X-Amzn-RequestId,x-amzn-alt-domain,Date,Location
Access-Control-Allow-Credentials: true
Date: Thu, 29 Nov 2018 00:53:19 GMT
Connection: close
```

**Here is the output for trying to speak:**
```
core-ssh:/config/alexa# bash alexa_remote_control.sh -d 'Jeffs Dot' -e speak:'test'                             sending cmd:speak:test to dev:Jeffs Dot type:A3S5BH2HU6VAYF serial:G090L90965222ET4 customerid:A17XZBYWJG1SFN
Sequence command: Alexa.Speak
HTTP/1.1 0 x-amzn-RequestId: 645201df-f371-11e8-bb41-53424d01fabd
Access-Control-Allow-Origin: https://alexa.amazon.com
Content-Encoding: deflate
Vary: Origin
Access-Control-Expose-Headers: x-amzn-dat-gui-client-upgrade,X-Amzn-Error,loginUrl,X-Amzn-RequestId,x-amzn-alt-domain,Date,Location
Access-Control-Allow-Credentials: true
Date: Thu, 29 Nov 2018 00:54:57 GMT
Connection: close
```
**I am new to alexa-remote-control.sh so perhaps its something I'm doing something wrong?**

**Here is output from successful -lastalexa query:**

```
core-ssh:/config/alexa# bash alexa_remote_control.sh -lastalexa
Bathroom
core-ssh:/config/alexa#

```

**I'm in the US:**

```
LANGUAGE="en-US"

AMAZON='amazon.com'

ALEXA='pitangui.amazon.com'

```

**Comment by PlayedIn** [comment]
I'm in the US:

```
LANGUAGE="en-US"

AMAZON='amazon.com'

ALEXA='pitangui.amazon.com'

```

**Comment by Apollon77** [comment]
Can you trigger the same action using the Alexa app on your mobile phone? Is it working when you start such a routine in the app?

**Comment by PlayedIn** [comment]
Yes just now tried it.  The routine is "Open Door".  When I triggered it just now on the Alexa app on my phone, the garage door opened.

**Comment by adn77** [comment]
I have to check if device names containing spaces are properly escaped...
Will get back later today.

**Comment by PlayedIn** [comment]
Thank you! Please know that it also doesn't work on my echoes that have no space.  I have a dot called garage and also one called bathroom.  See below:

```
core-ssh:/config/alexa# bash alexa_remote_control.sh -d 'Bathroom' -e automation:'Open Door'
sending cmd:automation:Open Door to dev:Bathroom type:A3S5BH2HU6VAYF serial:G090LF1073140NQA customerid:A17XZBYWJG1SFN
Running routine: Open Door
HTTP/1.1 0 x-amzn-RequestId: 69a7ee2e-f3ea-11e8-91f5-3185f340f4ee
Access-Control-Allow-Origin: https://alexa.amazon.com
Content-Encoding: deflate
Vary: Origin
Access-Control-Expose-Headers: x-amzn-dat-gui-client-upgrade,X-Amzn-Error,loginUrl,X-Amzn-RequestId,x-amzn-alt-domain,Date,Location
Access-Control-Allow-Credentials: true
Date: Thu, 29 Nov 2018 15:21:15 GMT
Connection: close

core-ssh:/config/alexa# bash alexa_remote_control.sh -d 'Garage' -e speak:'This is a test'
sending cmd:speak:This is a test to dev:Garage type:A3S5BH2HU6VAYF serial:G090LF0964860A5C customerid:A17XZBYWJG1SFN
Sequence command: Alexa.Speak
HTTP/1.1 0 x-amzn-RequestId: 98ddceed-f3ea-11e8-a77f-f3d91584ddf5
Access-Control-Allow-Origin: https://alexa.amazon.com
Content-Encoding: deflate
Vary: Origin
Access-Control-Expose-Headers: x-amzn-dat-gui-client-upgrade,X-Amzn-Error,loginUrl,X-Amzn-RequestId,x-amzn-alt-domain,Date,Location
Access-Control-Allow-Credentials: true
Date: Thu, 29 Nov 2018 15:22:34 GMT
Connection: close
```

**Comment by PlayedIn** [comment]
Cookie expired yesterday so my Hassio sensor.last_alexa stopped working.   Tried to recreate the cookie the way I did initially in Firefox.  Deleted all Firefox cookies, logged in successfully to Alexa.Amazon.com.   Exported the cookie.  Copied to same folder as Alexa_remote_control.sh.   Ran the script but it kept saying the cookie was expired.  

Recreated the cookie several times, always said it was expired.  Enabled 2FA on Amazon.  In Firefox, logged in to create the cookie.  Exported it and the script worked with this cookie. 

Then tried again to "speak" and also 'run a routine'.  Both worked!  Not sure why everything is working now.   Any thoughts on this before I close this issue?
